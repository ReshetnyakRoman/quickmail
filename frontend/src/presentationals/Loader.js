import React from 'react'

export default function Loader (props) {
  return (
    <div className='my-loader-container d-flex'>
      <div className='loader'>
        <span>Loading...</span>
      </div>
    </div>
  )
}
