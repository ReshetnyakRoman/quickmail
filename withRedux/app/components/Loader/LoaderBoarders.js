import styled, { keyframes } from 'styled-components';
import { withTheme } from '@callstack/react-theme-provider';

const rotating = keyframes`
  50% {
    -webkit-transform: rotate(calc(180deg * var(--direction)));
    -moz-transform: rotate(calc(180deg * var(--direction)));
    -o-transform: rotate(calc(180deg * var(--direction)));
    -ms-transform: rotate(calc(180deg * var(--direction)));
    transform: rotate(calc(180deg * var(--direction)));
  }
  100% {
    -webkit-transform: rotate(calc(360deg * var(--direction)));
    -moz-transform: rotate(calc(360deg * var(--direction)));
    -o-transform: rotate(calc(360deg * var(--direction)));
    -ms-transform: rotate(calc(360deg * var(--direction)));
    transform: rotate(calc(360deg * var(--direction)));
  }
`;

const LoaderBoarders = styled.div`
  width: 10rem;
  height: 10rem;
  font-size: 25px;
  border-top: 0.3rem solid #ff69b4;
  border-radius: 50%;
  position: relative;
  --direction: 1;
  -webkit-animation: ${rotating} 2s ease-in-out infinite;
  -moz-animation: ${rotating} 2s ease-in-out infinite;
  -o-animation: ${rotating} 2s ease-in-out infinite;
  -ms-animation: ${rotating} 2s ease-in-out infinite;
  animation: ${rotating} 2s ease-in-out infinite;
  background-color: ${props => props.theme.blurLayerColor};

  ::before,
  ::after {
    content: '';
    position: absolute;
    width: inherit;
    height: inherit;
    border-radius: 50%;
    top: -0.2rem;
  }
  ::before {
    border-top: 0.3rem solid #1e90ff;
    -webkit-transform: rotate(120deg);
    -moz-transform: rotate(120deg);
    -o-transform: rotate(120deg);
    -ms-transform: rotate(120deg);
    transform: rotate(120deg);
  }
  ::after {
    border-top: 0.3rem solid #ffd700;
    -webkit-transform: translateX(-100%) rotate(240deg);
    -moz-transform: translateX(-100%) rotate(240deg);
    -o-transform: translateX(-100%) rotate(240deg);
    -ms-transform: translateX(-100%) rotate(240deg);
    transform: translateX(-100%) rotate(240deg);
  }
  & span {
    display: inline-block;
    postion: absolute;
    color: ${props => props.theme.primaryTextColor};
    width: inherit;
    height: inherit;
    text-align: center;
    line-height: 10rem;
    font-size: 16px;
    -webkit-animation: ${rotating} 2s linear infinite;
    -moz-animation: ${rotating} 2s linear infinite;
    -o-animation: ${rotating} 2s linear infinite;
    -ms-animation: ${rotating} 2s linear infinite;
    animation: ${rotating} 2s linear infinite;
    --direction: -1;
  }
`;

export default withTheme(LoaderBoarders);
