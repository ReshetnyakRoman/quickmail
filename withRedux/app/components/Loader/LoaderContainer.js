import styled from 'styled-components';
import { withTheme } from '@callstack/react-theme-provider';

const LoaderContainer = styled.div`
  width: 100%;
  height: 100%;
  top: 0%;
  left: 0%;
  padding: 0;
  z-index: 50;
  position: fixed;
  -webkit-box-pack: center;
  -moz-box-pack: center;
  -o-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-box-align: center;
  -moz-box-align: center;
  -o-box-align: center;
  -ms-flex-align: center;
  -webkit-align-items: center;
  align-items: center;
  background-color: ${props => props.theme.blurLayerColor};
  display: flex;
`;

export default withTheme(LoaderContainer);
