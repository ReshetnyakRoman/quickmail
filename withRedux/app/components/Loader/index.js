import React from 'react';
import LoaderBoarders from './LoaderBoarders';
import LoaderContainer from './LoaderContainer';

const Loader = () => (
  <LoaderContainer>
    <LoaderBoarders>
      <span>Loading...</span>
    </LoaderBoarders>
  </LoaderContainer>
);

export default Loader;
