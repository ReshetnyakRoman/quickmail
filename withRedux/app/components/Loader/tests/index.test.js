import React from 'react';
import renderer from 'react-test-renderer';
import Loader from '../index';

describe('<Loader />', () => {
  it('should match snapshot', () => {
    const rendredComponent = renderer.create(<Loader />).toJSON();
    expect(rendredComponent).toMatchSnapshot();
  });
});
