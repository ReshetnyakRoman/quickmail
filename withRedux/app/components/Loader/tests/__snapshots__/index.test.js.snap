// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loader /> should match snapshot 1`] = `
<div
  className="LoaderContainer-pwzsga-0 fjBqNb"
>
  <div
    className="LoaderBoarders-zole2k-0 bNDcYJ"
  >
    <span>
      Loading...
    </span>
  </div>
</div>
`;
